jenkins/bootstrap_test.py

```diff
@@ -1133,6 +1133,14 @@ def testAllJobsHaveErrExit(self):
                      expected, lines,
                      '%s not found in %s' % (expected, job_path))
 
+    def testNoBadVarsInJobs(self):
+        """Searches for jobs that contain ${{VAR}}"""
+        for job, job_path in self.jobs:
+            with open(job_path) as fp:
+                script = fp.read()
+            bad_vars = re.findall("(\${{.+}})", script)
```

Comment:
`\$` is not a valid character, and the fact that python translates this to `\\$` is a lucky coincidence. We need the `r'\$'` so that you tell python not to treat backslash as an escape character in the string literal so it can be used as an escape character inside the regex.